Kitodo/src/main/java/org/kitodo/production/services/data/TaskService.java

```diff
@@ -685,84 +687,26 @@ public List<Task> getPreviousTasksForProblemReporting(Integer ordering, Integer
-     */
-    List<Map<String, Object>> findByTemplateId(Integer id) throws DataException {
-        return findDocuments(getQueryForTemplate(id));
-    }
-
-    /**
-     * Get query for automatic type of task.
-     *
-     * @param typeAutomatic
-     *            automatic type of task as boolean
-     * @return query as QueryBuilder
-     */
-    private QueryBuilder getQueryForTypeAutomatic(boolean typeAutomatic) {
-        return createSimpleQuery(TaskTypeField.TYPE_AUTOMATIC.getKey(), typeAutomatic, true);
-    }
-
-    /**
-     * Get query for process.
-     *
-     * @param processId
-     *            process id as int
-     * @return query as QueryBuilder
-     */
-    private QueryBuilder getQueryForProcess(int processId) {
-        return createSimpleQuery(TaskTypeField.PROCESS_ID.getKey(), processId, true);
-    }
-
-    /**
-     * Get query for processing user.
-     *
-     * @param processingUserId
-     *            processing user id as int
-     * @return query as QueryBuilder
-     */
-    private QueryBuilder getQueryForProcessingUser(int processingUserId) {
-        return createSimpleQuery(TaskTypeField.PROCESSING_USER_ID.getKey(), processingUserId, true);
-    }
-
-    /**
-     * Get query for processing status.
-     *
-     * @param processingStatus
-     *            processing status as int
-     * @return query as QueryBuilder
-     */
-    private QueryBuilder getQueryForProcessingStatus(int processingStatus) {
-        return createSimpleQuery(TaskTypeField.PROCESSING_STATUS.getKey(), processingStatus, true);
-    }
-
-    /**
-     * Get query for processing statuses.
-     *
-     * @param processingStatus
-     *            set of processing statuses as Integer
-     * @return query as QueryBuilder
-     */
-    private QueryBuilder getQueryForProcessingStatuses(Set<Integer> processingStatus) {
-        return createSetQuery(TaskTypeField.PROCESSING_STATUS.getKey(), processingStatus, true);
-    }
-
-    /**
-     * Get query for template.
+     * Determines all processes with a specific production template.
      *
      * @param templateId
-     *            template id as int
-     * @return query as QueryBuilder
+     *            record number of the production template
+     * @return list that is not empty if something was found, otherwise empty
+     *         list
+     * @throws DAOException
+     *             if an error occurred during the search
      */
-    private QueryBuilder getQueryForTemplate(int templateId) {
-        return createSimpleQuery(TaskTypeField.TEMPLATE_ID.getKey(), templateId, true);
+    /*
+     * Used in TemplateForm to find out whether a production template is used in
+     * a process. (Then it may not be deleted.) Is only checked for isEmpty().
+     */
+    public List<Map<String, Object>> findByTemplateId(Integer id) throws DAOException {
+        throw new UnsupportedOperationException("not yet implemented");
```

Comment:
Method is not marked as deprecated but is throwing this exception? Why not remove this method and maybe adjust the still used usage places?